From 1cf1ca9467d84882daea5acf7426b214dd0a7fee Mon Sep 17 00:00:00 2001
From: Bin Lan <bin.lan.cn@windriver.com>
Date: Mon, 9 Jun 2025 14:51:50 +0800
Subject: [PATCH 4/4] fs/yaffs2: fix a build warning in yaffs_mtdif.c when
 running make allyesconfig
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When building linux yocto with allyesconfig, a build warning is
found:

fs/yaffs2/yaffs_mtdif.c:48:5: error: no previous prototype for
‘nandmtd_erase_block’ [-Werror=missing-prototypes]
   48 | int nandmtd_erase_block(struct yaffs_dev *dev, int block_no)
      |     ^~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Inspect the code that the function nandmtd_erase_block() is
not used by yaffs2. So remove it.

Signed-off-by: Bin Lan <bin.lan.cn@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 fs/yaffs2/yaffs_mtdif.c | 28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git a/fs/yaffs2/yaffs_mtdif.c b/fs/yaffs2/yaffs_mtdif.c
index ecc53f6814b6..1c0d895e7885 100644
--- a/fs/yaffs2/yaffs_mtdif.c
+++ b/fs/yaffs2/yaffs_mtdif.c
@@ -45,34 +45,6 @@
 #define mtd_block_markbad(m, offs) (m)->block_markbad(m, offs)
 #endif
 
-int nandmtd_erase_block(struct yaffs_dev *dev, int block_no)
-{
-	struct mtd_info *mtd = yaffs_dev_to_mtd(dev);
-	u32 addr = ((loff_t) block_no) * dev->param.total_bytes_per_chunk *
-		dev->param.chunks_per_block;
-	struct erase_info ei;
-	int retval = 0;
-
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0))
-	ei.mtd = mtd;
-#endif
-	ei.addr = addr;
-	ei.len = dev->param.total_bytes_per_chunk * dev->param.chunks_per_block;
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0))
-	ei.time = 1000;
-	ei.retries = 2;
-	ei.callback = NULL;
-	ei.priv = (u_long) dev;
-#endif
-
-	retval = mtd_erase(mtd, &ei);
-
-	if (retval == 0)
-		return YAFFS_OK;
-
-	return YAFFS_FAIL;
-}
-
 static int yaffs_mtd_write(struct yaffs_dev *dev, int nand_chunk,
 			   const u8 *data, int data_len,
 			   const u8 *oob, int oob_len)
-- 
2.39.2

